recipes-extended/xen/files/hvmloader-legacy-seabios-optionroms.patch

```diff
@@ -122,10 +140,28 @@ PATCHES
  struct seabios_info {
      char signature[14]; /* XenHVMSeaBIOS\0 */
      uint8_t length;     /* Length of this struct */
-@@ -141,10 +153,66 @@ static void seabios_load(const struct bi
-     seabios_config.image_size = bios_length;
+@@ -130,9 +142,11 @@ static void seabios_setup_e820(void)
+     dump_e820_table(e820, info->e820_nr);
  }
  
++static void *ipxe_module_addr;
```

Comment:
Is there a ticket number for the PXE related changes?  Or is this related to Xen 4.12 uprev?  Can this change go upstream?

Reply:
See https://github.com/xen-project/xen/commit/e4506f404cb7983d6b424054ba999c6dd26ab4fa for the related commit. The IPXE module is no longer embedded (it is now loaded as an external module), so we can't load the etherboot ROM using our current methodology. Instead, we have to store the address of the external module at the time of the load and reference it later. I mirrored their changes for the ROMBIOS here.

I don't believe this change can go upstream as upstream doesn't support building/loading SEABIOS ROMs from out-of-tree (i.e. the original purpose of this patch).